### DISCUSSION TITLE
How do I set the api key?

### DISCUSSION THREAD

**Discussion by amir16yp** [Christopher-Hayes/vscode-chatgpt-reborn#18] [Q&A] [answered]
its not the extension settings. i saw that you changed to use "secure storage", whatever that means, i cant find anything about it on google and not in the entire vscode settings. am i missing something? how do i set the damn api key???

**Comment by Christopher-Hayes** [comment, marked as answer]
**New users** - will be greeted with a screen to set the API key.

**Existing users** - your API key should've been automatically moved from settings.json to secure storage. If you want to change the api key, hit "More actions" > "Reset API key"

![image](https://user-images.githubusercontent.com/6013871/228262834-a16973c7-9519-4192-9ec3-b59b7fbbaafb.png)
